Line Ending Conversion Correction

bwmilby opened the pull request:
Native Mac line endings are not being converted (CRLF and LFCR are converted to LF but CR was not touched).
Changed to use the LF constant to make it more readable.
Also changed numToChar to numToCodepoint since the former function is deprecated.

bwmilby committed:
Line Ending Conversion Correction

Native Mac line endings are not being converted (CRLF and LFCR are converted to LF but CR was not touched).
Changed to use the LF constant to make it more readable.
Also changed numToChar to numToCodepoint since the former function is deprecated.

bwmilby committed:
Remove LFCR replacements

revidelibrary.8.livecodescript
	private function __escapeStringAndConvertLineEndings pString
	private function escape pString, pConvertLineEndings

montegoulding commented on Toolset/libraries/revidelibrary.8.livecodescript:
```diff
@@ -10679,10 +10679,10 @@ private function escape pString, pConvertLineEndings
-      replace (numToChar(13) & CR) with (CR & numToChar(13)) in pString
-      replace (CR & numToChar(13)) with CR in pString
+      replace CRLF with "\n" in pString
+      replace numToCodepoint(13) with "\n" in pString
```
@bwmilby would need to examine all the code paths leading here but I'm wondering if `pConvertLineEndings` is false here we should be escaping `numToCodepoint(13)` with `\r`

bwmilby replied:
@montegoulding Function only used once (line 11050) and it has "true" as the convert value.  It probably makes sense to put an else clause in though.

bwmilby committed:
Line Ending Conversion Correction

Add handling for the case that an orphan carriage return character is found and for when not converting line endings.

montegoulding commented:
@livecode-vulcan review ok e91b84f

livecode-vulcan commented:
💙 review by @montegoulding ok e91b84f4d57060250b2d0c78b0d3014ee6e6fd8f
